src/algebraic_topology/simplex_category.lean

```diff
@@ -402,4 +402,167 @@ full_subcategory_inclusion _
+            { transitivity b,
+              exact a_leq_b,
+              exact b_leq_x },
+            exact h a_leq_x },
+        simp [b_ge_x]}}⟩ with def_chi_1,
+  set chi_2 : m ⟶ [1] := hom.mk
+  ⟨λ u, if u < x then 0 else 1,
+   by { intros a b a_leq_b,
+        by_cases (a = b),
+        { rw h },
+        rename h a_neq_b,
+        have a_lt_b : a < b,
+        { cases nat.eq_or_lt_of_le a_leq_b,
+          { exfalso, exact a_neq_b (subtype.eq h) },
+          { exact h }},
+        by_cases a < x,
+        { simp [h], obviously },
+        { simp [h],
+          have b_geq_x : ¬(b < x),
+          { by_contra b_leq_x,
+            have a_lt_x : a < x,
+            { transitivity b,
+              exact a_lt_b,
+              exact b_leq_x },
+            exact h a_lt_x },
+          simp [b_geq_x] }}⟩ with def_chi_2,
+  -- The two auxiliary functions equalize f
+  have f_comp_chi_i : f ≫ chi_1 = f ≫ chi_2,
+  { dsimp,
+    ext,
+    rw [hom.to_preorder_hom_mk, hom.to_preorder_hom_mk, preorder_hom.comp_coe],
+    rw [hom.to_preorder_hom_mk, hom.to_preorder_hom_mk, preorder_hom.comp_coe],
+    rw [function.comp_app, function.comp_app],
+    have f_x_1_ne_x : ¬((f.to_preorder_hom) x_1 = x) := h_ab x_1,
+    by_cases ((f.to_preorder_hom) x_1) ≤ x,
+    { have f_x_1_lt_x : ((f.to_preorder_hom) x_1) < x,
+      { cases nat.eq_or_lt_of_le h,
+        exfalso,
+        replace h_1 := subtype.eq h_1,
+        exact f_x_1_ne_x h_1,
+        exact h_1 },
+      simp [f_x_1_lt_x, h] },
+    { have n_f_x_1_lt_x : ¬(((f.to_preorder_hom) x_1) < x),
+    { erw not_iff_not_of_iff nat.lt_iff_le_not_le,
+      rw not_and,
+      intro h3, exfalso,
+      refine h h3 },
+      simp [n_f_x_1_lt_x, h] }},
+  resetI,
+  -- We now just have to show the two auxiliary functions are not equal.
+  rw category_theory.cancel_epi f at f_comp_chi_i, rename f_comp_chi_i eq_chi_i,
+  apply_fun hom.to_preorder_hom at eq_chi_i,
+  apply_fun preorder_hom.to_fun at eq_chi_i,
+  replace eq_chi_i := congr_fun eq_chi_i x,
+  dsimp at eq_chi_i,
+  have chi_1_x : (hom.to_preorder_hom chi_1) x = 0 := by { simp },
+  have chi_2_x : (hom.to_preorder_hom chi_2) x = 1 := by { simp },
+  rw [chi_1_x, chi_2_x] at eq_chi_i,
+  refine nat.zero_ne_one (fin.veq_of_eq eq_chi_i) },
+  { intro hyp_surj,
+    refine ⟨ by { intros l g h h_eq_comp,
+      dsimp at *,
+      ext,
+      set y := Exists.some (hyp_surj x) with d_y,
+      have y_eq_f_x : x = f.to_preorder_hom y,
+       { rw d_y,
+        exact (Exists.some_spec (hyp_surj x)).symm },
+      apply_fun hom.to_preorder_hom at h_eq_comp,
+      apply_fun preorder_hom.to_fun at h_eq_comp,
+      replace h_eq_comp := congr_fun h_eq_comp y,
+      dsimp at h_eq_comp,
+      rw [hom.to_preorder_hom_mk, preorder_hom.comp_coe, function.comp_app] at h_eq_comp,
+      rw [hom.to_preorder_hom_mk, preorder_hom.comp_coe, function.comp_app] at h_eq_comp,
+      rw ←y_eq_f_x at h_eq_comp,
+      refine fin.veq_of_eq h_eq_comp }⟩}
```

Comment:
```suggestion
  { intro hyp_surj,
    exact concrete_category.epi_of_surjective _ hyp_surj }
```
like before, if you use that this category is concrete, this proof shortens a little :)

Reply:
Thank you a lot for suggesting the use of concrete categories, both for this theorem and the preceding! I thought an approach like this was possible, but did not know how to do it in lean. It simplifies a good part of the proof so let's use it.